### PR TITLE
Integrate Mayhem

### PR DESCRIPTION
**Pull request by rnshah9**

### CODE REVIEW

File: harness.c [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#include <stdint.h>
```
Comment (xansec) [inline harness.c]: It looks like we're not doing bounds checking here, so this bug: https://mayhem.forallsecure.com/rnshah9/elk/cli/-/defects/286085 appears to be in the harness itself. I would add bounds checking to the harness, (we want to find OOB reads in the underlying code, not the harness), and rename the harness to something that more closely resembles what's being fuzzed (in this case, js_eval).

**Review by xansec — CHANGES_REQUESTED** [review]
Please make the requested changes.

File: harness.c (hunk 1 above)
Comment (rnshah9) [reply harness.c]: I'm a bit confused; in the defect that you linked, the backtraces in the test cases all point to within `js_eval`, not outside of it. So aren't the OOB reads happening in the source code? e.g.
```
...
#19 0x4b2794 in js_stmt (/fuzz+0x4b2794)
#20 0x4b09fc in js_eval (/fuzz+0x4b09fc)
#21 0x4ae259 in LLVMFuzzerTestOneInput (/fuzz+0x4ae259)
#22 0x440a01 in fuzzer::Fuzzer::ExecuteCallback(unsigned char const*, unsigned long) (/fuzz+0x440a01)
#23 0x42c172 in fuzzer::RunOneTest(fuzzer::Fuzzer*, char const*, unsigned long) (/fuzz+0x42c172)
#24 0x431c26 in fuzzer::FuzzerDriver(int*, char***, int (*)(unsigned char const*, unsigned long)) (/fuzz+0x431c26)
#25 0x45a8e2 in main (/fuzz+0x45a8e2)
#26 0x7f5c1f997082 in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x24082)
#27 0x40683d in _start (/fuzz+0x40683d)
```

What bounds need to be checked? I don't see any array accesses in the harness, either.

Comment (xansec) [reply harness.c]: When I initially looked at this it seemed as though you were passing an incongruent length parameter to js_eval() (the `~0U`), which seemed disingenous. Now after passing the corresponding size parameter instead I am still seeing the crash pop up, so I think this is legitimate. This also seems to be the case after looking at the declaration here: 
https://github.com/cesanta/elk/blob/9112ce0f7e5ecfefa176d3fd827b17c84b720c00/elk.c#L1387

Anyways, I did make some other changes, which you can find in this PR: https://github.com/rnshah9/elk/pull/1

**Comment by xansec** [comment]
LGTM!

**Pull request merged**